api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/plugins/ContrastPluginServiceImpl.java

```diff
@@ -0,0 +1,175 @@
+        ContrastPluginRequest accountData = objectMapper.convertValue(pluginRequest, ContrastPluginRequest.class);
+        LOGGER.info(String.format(VALIDATING_MSG, pluginName));
+        PluginResponse validationResponse = validateContrastPluginRequest(accountData, pluginName);
+        if (validationResponse.getStatus().equalsIgnoreCase(AdminConstants.FAILURE)) {
+            LOGGER.info(VALIDATION_FAILED);
+            return validationResponse;
+        }
+        try {
+            String organizationName = getOrganizationNameFromApiResponse(accountData);
+            LOGGER.info("Organization name for orgId : {} is {}", accountData.getOrganizationId(), organizationName);
+            if (StringUtils.isEmpty(organizationName)) {
+                return new PluginResponse(AdminConstants.FAILURE, "Couldn't read Organization name",
+                        "Organization name is null, terminating sequence");
+            }
+            return new PluginResponse(AdminConstants.SUCCESS, VALIDATION_SUCCESSFUL, null);
+        } catch (PluginApiResponseException pare) {
+            LOGGER.error(VALIDATION_FAILED + pare.getMessage());
+            return new PluginResponse(AdminConstants.FAILURE, VALIDATION_FAILED, pare.getMessage());
+        } catch (Exception e) {
+            LOGGER.error(VALIDATION_FAILED + UNEXPECTED_ERROR_MSG, e.getMessage());
+            return new PluginResponse(AdminConstants.FAILURE, VALIDATION_FAILED, e.getMessage());
+        }
+    }
+
+    private PluginResponse validateContrastPluginRequest(ContrastPluginRequest pluginData, String pluginName) {
+        PluginResponse response = new PluginResponse();
+        StringBuilder validationErrorDetails = new StringBuilder();
+        if (StringUtils.isEmpty(pluginData.getOrganizationId())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "organizationId"));
+        }
+        if (StringUtils.isEmpty(pluginData.getServiceKey())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "serviceKey"));
+        }
+        if (StringUtils.isEmpty(pluginData.getApiKey())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "apiKey"));
+        }
+        if (StringUtils.isEmpty(pluginData.getUserId())) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, "userId"));
+        }
+        String validationError = validationErrorDetails.toString();
+        if (!validationError.isEmpty()) {
+            response.setStatus(AdminConstants.FAILURE);
+            response.setMessage(String.format(INVALID_PLUGIN_REQUEST_MSG, pluginName));
+            response.setErrorDetails(validationError);
+        } else {
+            response.setStatus(AdminConstants.SUCCESS);
+        }
+        return response;
+    }
+
+    public String getOrganizationNameFromApiResponse(ContrastPluginRequest accountData) {
```

Comment:
why it is public?
getOrganizationName is sufficient enough

Reply:
sorry, changed